soh/soh/ResourceManagerHelpers.cpp

```diff
@@ -503,3 +512,139 @@ extern "C" void ResourceMgr_ClearSkeletons() {
 extern "C" s32* ResourceMgr_LoadCSByName(const char* path) {
     return (s32*)ResourceMgr_GetResourceDataByNameHandlingMQ(path);
 }
+
+bool IsSharedScene(int16_t sceneNum) {
+    return sceneNum > SCENE_INSIDE_GANONS_CASTLE || sceneNum == SCENE_GANONS_TOWER || sceneNum == SCENE_THIEVES_HIDEOUT;
```

Comment:
i think being explicit here could be nice. listing out all the scenes that aren't shared makes it easier to tell from a glance what "shared" means, as opposed to leading someone unfamiliar to go digging into `scene_table.h` to understand what this is checking for.

it's a bit more verbose (11 lines) this way but it doesn't feel bad imo

```suggestion
    return sceneNum != SCENE_DEKU_TREE &&
           sceneNum != SCENE_DODONGOS_CAVERN &&
           sceneNum != SCENE_JABU_JABU &&
           sceneNum != SCENE_FOREST_TEMPLE &&
           sceneNum != SCENE_FIRE_TEMPLE &&
           sceneNum != SCENE_WATER_TEMPLE &&
           sceneNum != SCENE_SPIRIT_TEMPLE &&
           sceneNum != SCENE_SHADOW_TEMPLE &&
           sceneNum != SCENE_BOTTOM_OF_THE_WELL &&
           sceneNum != SCENE_ICE_CAVERN &&
           sceneNum != SCENE_GERUDO_TRAINING_GROUND;
```